File: backend/app/repositories/plantas.py
"""Repositorio para plantas"""
from sqlalchemy.orm import Session
from typing import List, Optional
from ..models import Planta
from ..schemas import PlantaCreate, PlantaUpdate


class PlantaRepository:
    """Repositorio para operaciones CRUD de plantas"""
    
    def get_all(self, db: Session) -> List[Planta]:
        """Obtener todas las plantas"""
        return db.query(Planta).order_by(Planta.nombre).all()
    
    def get_by_id(self, db: Session, id_planta: int) -> Optional[Planta]:
        """Obtener planta por ID"""
        return db.query(Planta).filter(Planta.id_planta == id_planta).first()
    
    def get_by_codigo(self, db: Session, codigo: str) -> Optional[Planta]:
        """Obtener planta por código"""
        return db.query(Planta).filter(Planta.codigo == codigo).first()
    
    def create(self, db: Session, planta_data: PlantaCreate) -> Planta:
        """Crear nueva planta"""
        planta = Planta(**planta_data.model_dump())
        db.add(planta)
        db.commit()
        db.refresh(planta)
        return planta
    
    def update(self, db: Session, planta: Planta, planta_data: PlantaUpdate) -> Planta:
        """Actualizar planta existente"""
        update_data = planta_data.model_dump(exclude_unset=True)
        for key, value in update_data.items():
            setattr(planta, key, value)
        db.commit()
        db.refresh(planta)
        return planta
    
    def delete(self, db: Session, planta: Planta) -> None:
        """Eliminar planta"""
        db.delete(planta)
        db.commit()


planta_repository = PlantaRepository()

File: frontend/src/api/plantas.ts
/**
 * Servicio API para plantas
 */
import axios from './axios';
import type { Planta, PlantaCreate, PlantaUpdate, ApiMessage } from '../types';

export const plantasApi = {
  /**
   * Obtener todas las plantas
   */
  listar: async (): Promise<Planta[]> => {
    const response = await axios.get<Planta[]>('/plantas');
    return response.data;
  },

  /**
   * Crear nueva planta
   */
  crear: async (data: PlantaCreate): Promise<Planta> => {
    const response = await axios.post<Planta>('/plantas', data);
    return response.data;
  },

  /**
   * Actualizar planta
   */
  actualizar: async (id: number, data: PlantaUpdate): Promise<Planta> => {
    const response = await axios.put<Planta>(`/plantas/${id}`, data);
    return response.data;
  },

  /**
   * Eliminar planta
   */
  eliminar: async (id: number): Promise<ApiMessage> => {
    const response = await axios.delete<ApiMessage>(`/plantas/${id}`);
    return response.data;
  },
};
